### PR TITLE
Documentation update s8 ultra

### PR DESCRIPTION
**Pull request by chrisn-au**
Gents, I have done a bit of work to restructure the documentation and provide the basis to extend from. Thank you for the support to get to this point.

The documentation builds cleanly
```
autosummary] generating autosummary for: api_commands.rst, error.rst, index.rst, status.rst, usage.rst
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from https://www.sphinx-doc.org/en/master/objects.inv...
building [mo]: targets for 0 po files that are out of date
writing output... 
building [html]: targets for 5 source files that are out of date
updating environment: [new config] 5 added, 0 changed, 0 removed
reading sources... [100%] usage
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
copying assets... copying static files... done
copying extra files... done
done
writing output... [100%] usage
generating indices... genindex done
writing additional pages... search done
dumping search index in English (code: en)... done
dumping object inventory... done

====================== slowest reading durations =======================
0.285 api_commands
0.009 error
0.009 status
0.005 usage
0.004 index
build succeeded.
```

and is grouped by logical command grouping (acknowledging that this is subjective)

I will continue to make changes (tested on my Roborock s8 Pro Ultra) however will wait until this pull request is approved

Feel free to reach out with questions

### CODE REVIEW

File: docs/source/api_commands.rst [hunk 1]
```diff
@@ -47,183 +390,340 @@ Return example::
...
+~~~~~~~~~~~
+
+Description: Moves the robot in the direction specified
+
+Parameters: To be documented
+
+Returns ok or error
+
+..
+    Need to document the parameters - will need to explore the app to find out what they are
```
Comment (Lash-L) [inline docs/source/api_commands.rst:447]: all rc is partially implemented here: https://github.com/humbertogontijo/homeassistant-roborock/blob/main/custom_components/roborock/vacuum.py

If you look at the functions with "manual" in them.

Comment (Lash-L) [reply docs/source/api_commands.rst:447]: if that helps

File: docs/source/api_commands.rst [hunk 2]
```diff
@@ -47,183 +390,340 @@ Return example::
...
+app_zoned_clean
+~~~~~~~~~~~~~~~
 
-Description:
+Description: Starts a zone clean
 
 Parameters:
 
+.. 
+    Us this the last known zone
```
Comment (Lash-L) [inline docs/source/api_commands.rst:529]: iirc - i odn't use zoned cleans, but it should be able to accept a zone id.

Comment (Lash-L) [reply docs/source/api_commands.rst:529]: Shouldn't this be down in the following section?

File: docs/source/api_commands.rst [hunk 3]
```diff
@@ -276,901 +774,1168 @@ Return example::
...
-
-Set: set_carpet_mode
+set_carpet_mode
+~~~~~~~~~~~~~~~~~~~~
 
 Description:
 
 Parameters:
 
+get_smart_wash_params
```
Comment (Lash-L) [inline docs/source/api_commands.rst:790]: duplicated at 814

Comment (Lash-L) [reply docs/source/api_commands.rst:790]: smart wash is a dock specific command - i believe it has to do with cleaning the mop pads, idk what it does exactly as i don't have a facny dock

File: docs/source/api_commands.rst [hunk 4]
```diff
@@ -276,901 +774,1168 @@ Return example::
...
+
+set_identify_ground_material_status
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Description:
 
 Parameters:
 
+..
+    Method not known for S8 Pro Ultra
```
Comment (Lash-L) [inline docs/source/api_commands.rst:1094]: should match the other format with the chart and say not supported for s8 pro ultra.

File: docs/source/api_commands.rst [hunk 5]
```diff
@@ -276,901 +774,1168 @@ Return example::
...
 
-    clean_count:
+set_led_status
+~~~~~~~~~~~~~~
 
-    dust_collection_count:
+Description:  Sets the LED status. If disabled the indicator light will turn off 1 minute after fully charged
 
-    records:
+Parameters: ????
```
Comment (Lash-L) [inline docs/source/api_commands.rst:1139]: i believe just 1 and 0. I do it in core integration but i dont recall right now

File: docs/source/api_commands.rst [hunk 6]
```diff
@@ -276,901 +774,1168 @@ Return example::
...
+----
+
+get_multi_map
+~~~~~~~~~~~~~
 
 Description:
 
 Parameters:
 
+Comment: Response timed out for S8 Pro Ultra
```
Comment (Lash-L) [inline docs/source/api_commands.rst:1157]: needs to be done with the cloud client i think. but that may just be some of the other map commands, I haven't used this one

File: docs/source/api_commands.rst [hunk 7]
```diff
@@ -276,901 +774,1168 @@ Return example::
...
-Parameters:
+Parameters: Unknown
 
+Comment: Returns a map in a format that is not yet understood by me
 
-get_mop_template_params_summary
--------------------------------
+..
+    Explore what parameters it may take
+    Extend code to return byte stream ?
```
Comment (Lash-L) [inline docs/source/api_commands.rst:1209]: we do this through an external package

https://github.com/PiotrMachowski/Python-package-vacuum-map-parser-roborock

File: docs/source/api_commands.rst [hunk 8]
```diff
@@ -276,901 +774,1168 @@ Return example::
...
 
 Parameters:
 
 
-get_multi_map
--------------
+get_room_mapping
+~~~~~~~~~~~~~~~~
+
+Description: Returns a list of rooms, ids as discovered by
```
Comment (Lash-L) [inline docs/source/api_commands.rst]: important note is that this command is map specific. different maps give different results.

As well, you have more ids than I typically do, which doesn't make much sense. Mine are typically in groups of 2, i'm not sure what your third one is. 12,2, 9, 13,14

File: docs/source/api_commands.rst [hunk 9]
```diff
@@ -276,901 +774,1168 @@ Return example::
...
+Roborock S8 Pro Ultra   Yes
+======================  =========
+
+load_multi_map
+~~~~~~~~~~~~~~
 
 Description:
 
-Parameters:
+Parameters: ???
```
Comment (Lash-L) [inline docs/source/api_commands.rst:1246]: this is the same as selecting a map in the app

File: docs/source/api_commands.rst [hunk 10]
```diff
@@ -276,901 +774,1168 @@ Return example::
...
+{'sid_in_progress': 0, 'progress': 0, 'state': 0, 'error': 0}
 
-    collision_avoid_status:
+..
+    Is this where the vacumm is currently located ?
 
-    switch_map_mode:
 
-    dock_error_status:
+get_current_sound
```
Comment (Lash-L) [inline docs/source/api_commands.rst:1805]: i believ this one and the one above is where you change your voice/ download a new voice

**Review by Lash-L — APPROVED** [review]
Looking good - some reviews that can be addressed in a follow up. Merging this for now just to help with readability and changes

**Comment by Lash-L** [comment]
oh was gonna merge - but could you run pre-commit.

Ignore the lint commit message but the other lint needs to pass

File: docs/source/api_commands.rst (hunk 1 above)
Comment (chrisn-au) [reply docs/source/api_commands.rst:447]: nice thanks will have a look.

File: docs/source/api_commands.rst (hunk 6 above)
Comment (chrisn-au) [reply docs/source/api_commands.rst:1157]: Got it - an area to explore later

File: docs/source/api_commands.rst (hunk 7 above)
Comment (chrisn-au) [reply docs/source/api_commands.rst:1209]: My real reason for getting involved -thx

**Review by Lash-L — APPROVED** [review]

**Comment by Lash-L** [comment]
Forgot to change the commit title, but should be okay with my next merge

**Pull request merged**